nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/pgp/PGPControllerService.java

```diff
@@ -0,0 +1,893 @@
+                    } else {
+                        throw new PGPException("Encrypted data packet does not have integrity check.");
+                    }
+                }
+            } catch (final Exception e) {
+                throw new IOException(e.getMessage());
+            }
+        }
+    }
+
+
+    /**
+     * Read from an {@link InputStream} to generate a signature written to an {@link OutputStream}.
+     *
+     * @param input clear or cipher data
+     * @param signature to receive signature
+     * @param options used to configure sign operation
+     */
+    public void sign(InputStream input, OutputStream signature, SignOptions options) throws IOException, PGPException {
+        if (options == null || options.getPrivateKey() == null) {
+            throw new IOException("Sign operation invalid without Private Key");
+        }
+        final int KEY_ALGORITHM = 1; // wha
+        JcaPGPContentSignerBuilder builder = new JcaPGPContentSignerBuilder(KEY_ALGORITHM, options.getHashAlgorithm()).setProvider("BC");
+        PGPSignatureGenerator generator = new PGPSignatureGenerator(builder);
+        generator.init(PGPSignature.BINARY_DOCUMENT, options.getPrivateKey());
+        readAndUpdate(input, generator);
+        generator.generate().encode(signature);
+    }
+
+
+    /**
+     * Read from an {@link InputStream} to verify its signature.
+     *
+     * @param input clear or cipher data
+     * @param signature signature data
+     * @param options used to configure verify operation
+     * @return true if the signature matches
+     */
+    public boolean verify(InputStream input, InputStream signature, VerifyOptions options) throws IOException, PGPException {
+        JcaPGPObjectFactory signatureFactory = new JcaPGPObjectFactory(org.bouncycastle.openpgp.PGPUtil.getDecoderStream(signature));
+        PGPSignatureList signatures;
+
+        Object head = signatureFactory.nextObject();
+        if (head instanceof PGPCompressedData) {
+            PGPCompressedData compressed = (PGPCompressedData) head;
+            signatureFactory = new JcaPGPObjectFactory(compressed.getDataStream());
+            signatures = (PGPSignatureList) signatureFactory.nextObject();
+        } else if (head instanceof  PGPSignatureList) {
+            signatures = (PGPSignatureList) head;
+        } else {
+            throw new PGPException("Not a signature stream.");
+        }
+
+        PGPSignature innerSignature = signatures.get(0);
+        innerSignature.init(new JcaPGPContentVerifierBuilderProvider().setProvider("BC"), options.getPublicKey());
+        readAndUpdate(input, innerSignature);
+        return innerSignature.verify();
+    }
+
+
+    /**
+     * Generate an {@link EncryptOptions} instance for an encrypt operation.
+     *
+     * @param algorithm encryption algorithm identifier, see {@link org.bouncycastle.bcpg.SymmetricKeyAlgorithmTags}
+     * @param armor if true, encrypted data will be PGP-encoded ascii
```

Comment:
If `false`, what form is the output in?

Reply:
Added text to clarify that bytes are left unencoded after encryption when "armor" is false.